interfaces/api/pipeline/IImageSegmentationPipeline.h

```diff
@@ -0,0 +1,101 @@
+/**
+ * @copyright Copyright (c) 2025 B-com http://www.b-com.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SOLAR_IMAGESEGMENTATIONPIPELINE_H
+#define SOLAR_IMAGESEGMENTATIONPIPELINE_H
+
+#include "api/pipeline/IPipeline.h"
+#include "datastructure/Image.h"
+#include "datastructure/Mask2DCollection.h"
+#include <xpcf/core/helpers.h>
+
+namespace SolAR {
+using namespace datastructure;
+namespace api {
+namespace pipeline {
+
+/**
+ * @enum class ImageSegmentationStatus
+ */
+enum class ImageSegmentationStatus {
+    UNINITIALIZED = 0,    // processing not initialized
+    INITIALIZED = 1,      // processing correctly initialized, but not started
+    IN_PROGRESS = 2,      // processing in progress
+    COMPLETED = 3,        // processing completed
+    ABORTED = 4           // processing aborted before completion
+};
+
+/// @brief image segmentation status to string
+static const std::map<ImageSegmentationStatus, std::string> imageSegmentationStatusToString = {
+    {ImageSegmentationStatus::UNINITIALIZED, "UNINITIALIZED"},
+    {ImageSegmentationStatus::INITIALIZED, "INITIALIZED"},
+    {ImageSegmentationStatus::IN_PROGRESS, "IN_PROGRESS"},
+    {ImageSegmentationStatus::COMPLETED, "COMPLETED"},
+    {ImageSegmentationStatus::ABORTED, "ABORTED"}
+};
+
```

Comment:
Not consistent with "toString" methods defined in other pipelines (cf IMapProcessingPipeline for exemple)